### PR TITLE
build(deps): bump libvterm to 0.2 (#16217)

### LINKED ISSUE #16217: Build failed with Homebrew on both macOS 11.4 and 12.0.1
### Neovim version (nvim -v)

Upgrading neovim  HEAD-487286b -> HEAD-7899c40

### Vim (not Nvim) behaves the same?

not related

### Operating system/version

macOS 12.0.1

### Terminal name/version

alacritty 0.9.0 (fed349a)

### $TERM environment variable

alacritty

### Installation

homebrew

### How to reproduce the issue

```shell
brew install  neovim --HEAD
```
or
```shell
brew upgrade --fetch-HEAD
```

### Expected behavior

Install correctly like yesterday.

### Actual behavior

https://github.com/Homebrew/homebrew-core/issues/88665

```
==> make install
Last 15 lines from /Users/me/Library/Logs/Homebrew/neovim/04.make:
[ 50%] Building C object src/nvim/CMakeFiles/nvim.dir/ui.c.o
cd /tmp/neovim-20211103-75787-zm3uzo/build/src/nvim && /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang -DINCLUDE_GENERATED_DECLARATIONS -DMIN_LOG_LEVEL=3 -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -Dnvim_EXPORTS -I/tmp/neovim-20211103-75787-zm3uzo/build/config -I/tmp/neovim-20211103-75787-zm3uzo/src -I/tmp/neovim-20211103-75787-zm3uzo/build/src/nvim/auto -I/tmp/neovim-20211103-75787-zm3uzo/build/include -I/usr/local/opt/luajit-openresty/include/luajit-2.1 -isystem /usr/local/opt/gettext/include -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_TS_HAS_SET_MATCH_LIMIT -O2 -DNDEBUG -flto=thin -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=auto -MD -MT src/nvim/CMakeFiles/nvim.dir/ui.c.o -MF CMakeFiles/nvim.dir/ui.c.o.d -o CMakeFiles/nvim.dir/ui.c.o -c /tmp/neovim-20211103-75787-zm3uzo/src/nvim/ui.c
/tmp/neovim-20211103-75787-zm3uzo/src/nvim/terminal.c:818:29: error: passing 'VTermStringFragment' to parameter of incompatible type 'char *'
    buf_set_term_title(buf, val->string);
                            ^~~~~~~~~~~
/tmp/neovim-20211103-75787-zm3uzo/src/nvim/terminal.c:789:50: note: passing argument to parameter 'title' here
static void buf_set_term_title(buf_T *buf, char *title)
                                                 ^
1 error generated.
[ 50%] Building C object src/nvim/CMakeFiles/nvim.dir/ui_bridge.c.o
cd /tmp/neovim-20211103-75787-zm3uzo/build/src/nvim && /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang -DINCLUDE_GENERATED_DECLARATIONS -DMIN_LOG_LEVEL=3 -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -Dnvim_EXPORTS -I/tmp/neovim-20211103-75787-zm3uzo/build/config -I/tmp/neovim-20211103-75787-zm3uzo/src -I/tmp/neovim-20211103-75787-zm3uzo/build/src/nvim/auto -I/tmp/neovim-20211103-75787-zm3uzo/build/include -I/usr/local/opt/luajit-openresty/include/luajit-2.1 -isystem /usr/local/opt/gettext/include -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_TS_HAS_SET_MATCH_LIMIT -O2 -DNDEBUG -flto=thin -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=auto -MD -MT src/nvim/CMakeFiles/nvim.dir/ui_bridge.c.o -MF CMakeFiles/nvim.dir/ui_bridge.c.o.d -o CMakeFiles/nvim.dir/ui_bridge.c.o -c /tmp/neovim-20211103-75787-zm3uzo/src/nvim/ui_bridge.c
make[2]: *** [src/nvim/CMakeFiles/nvim.dir/terminal.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [src/nvim/CMakeFiles/nvim.dir/all] Error 2
make: *** [all] Error 2
```

### PR DESCRIPTION
**Pull request by erw7**
fixes #16217.

### CODE REVIEW

**Comment by nyngwang** [comment]
Thanks for your fixes! I'm waiting for this one.

**Comment by clason** [comment]
Shouldn't we update and use https://github.com/neovim/libvterm?

(See https://github.com/neovim/neovim/wiki/Deps#forks)

**Comment by clason** [comment]
Unfortunately, the update breaks terminal-related functional tests...

**Comment by jamessan** [comment]
> Unfortunately, the update breaks terminal-related functional tests...

The changes are more involved.  The vterm change means that we may not be getting the full escape sequence in one chunk.  We need to buffer the data until we get a fragment where `val->string.final` is true.

**Comment by jamessan** [comment]
> Shouldn't we update and use [neovim/libvterm](https://github.com/neovim/libvterm)?

Why? Release tarballs are different than the tagged commit.  There are things done during the release process that show up in the tarball.

**Comment by muniter** [comment]
This also closes https://github.com/neovim/neovim/issues/9555

**Comment by erw7** [comment]
`altscreen_spec.lua` will be broken in the next revision.

https://bazaar.launchpad.net/~libvterm/libvterm/trunk/revision/741

`buffer_spec.lua`, `window_spec.lua` will be broken in the next revision.

https://bazaar.launchpad.net/~libvterm/libvterm/trunk/revision/750

File: CMakeLists.txt [hunk 1]
```diff
@@ -466,9 +466,27 @@ if(FEAT_TUI)
   include_directories(SYSTEM ${LIBTERMKEY_INCLUDE_DIRS})
 endif()
 
-find_package(LIBVTERM 0.1 REQUIRED)
+find_package(LIBVTERM 0.1.4 REQUIRED)
 include_directories(SYSTEM ${LIBVTERM_INCLUDE_DIRS})
 
+list(APPEND CMAKE_REQUIRED_INCLUDES "${LIBVTERM_INCLUDE_DIRS}")
+list(APPEND CMAKE_REQUIRED_LIBRARIES "${LIBVTER_LIBRARIES}")
```
Comment (brcolow) [inline CMakeLists.txt:473]: Typo in "LIBVTER_LIBRARIES" should be "LIBVTERM_LIBRARIES"?

**Pull request closed without merging**